functions/vshn-postgres-func/encrypted_pvc.go

```diff
@@ -0,0 +1,72 @@
+package vshnpostgres
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/sethvargo/go-password/password"
+
+	"github.com/go-logr/logr"
+	"github.com/vshn/appcat-comp-functions/runtime"
+	vshnv1 "github.com/vshn/component-appcat/apis/vshn/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// AddPvcSecret adds a secret for the encrypted PVC for the  PostgreSQL instance.
+func AddPvcSecret(ctx context.Context, log logr.Logger, iof *runtime.Runtime, comp *vshnv1.VSHNPostgreSQL) (*vshnv1.VSHNPostgreSQL, error) {
+
+	log.Info("Check if encrypted storage is enabled")
+
+	log.V(1).Info("Transforming", "obj", iof)
+
+	// luksSecretResourceName is the resource name defined in the composition
+	// This name is different from metadata.name of the same resource
+	// The value is hardcoded in the composition for each resource and due to crossplane limitation
+	// it cannot be matched to the metadata.name
+	var luksSecretResourceName = comp.Name + "-luks-key"
+
+	encryptionSpec := comp.Spec.Parameters.Encryption
+
+	if _, err := strconv.ParseBool(encryptionSpec.Enabled); err == nil {
+
+		log.Info("Adding secret to composite")
+
+		s := &v1.Secret{}
+
+		err := iof.GetFromObservedKubeObject(ctx, s, luksSecretResourceName)
+		var luksKey string
+		if err != nil {
+			log.Info("Secret does not exist yet. Creating...")
+			luksKey, _ = password.Generate(64, 10, 1, false, true)
+		} else {
+			log.Info("Secret already exists. Use that one")
+			luksKey = string(s.Data["luksKey"])
+		}
+		log.Info("Foo: " + luksKey)
```

Comment:
Please don't log the secret

Reply:
Oh sorry. that was a leftover from some testing on my side